modules/transports/core/nhttp/src/test/java/org/apache/synapse/transport/passthru/TargetResponseTest.java

```diff
@@ -0,0 +1,148 @@
+        } catch (Exception e) {
+            logger.error(e);
+            Assert.fail("Unable to start the target response!");
+        }
+    }
+
+    /**
+     * Testing the starting of target response when response body is expected
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testTrue() throws Exception {
+        ConfigurationContext configurationContext = new ConfigurationContext(new AxisConfiguration());
+        WorkerPool workerPool = new NativeWorkerPool(3, 4, 5, 5, "name", "id");
+        PassThroughTransportMetricsCollector metrics = new PassThroughTransportMetricsCollector(true, "testScheme");
+
+        TargetConfiguration targetConfiguration = new TargetConfiguration(configurationContext, null, workerPool,
+                metrics, null);
+        targetConfiguration.build();
+        HttpResponse response = PowerMockito.mock(HttpResponse.class, Mockito.RETURNS_DEEP_STUBS);
+        NHttpClientConnection conn = PowerMockito.mock(NHttpClientConnection.class, Mockito.RETURNS_DEEP_STUBS);
+
+        PowerMockito.mockStatic(TargetContext.class);
+
+        TargetContext cntxt = new TargetContext(targetConfiguration);
+        PowerMockito.when(TargetContext.get(any(NHttpClientConnection.class))).thenReturn(cntxt);
+
+        TargetResponse targetResponse = new TargetResponse(targetConfiguration, response, conn, true, false);
+
+        try {
+            targetResponse.start(conn);
+        } catch (Exception e) {
+            logger.error(e);
+            Assert.fail("Unable to start the target response!");
+        }
+    }
+
+    /**
+     * Testing reading from a pipe
+     * @throws Exception
+     */
+    @Test
+    public void testRead() throws Exception {
+        ConfigurationContext configurationContext = new ConfigurationContext(new AxisConfiguration());
+        WorkerPool workerPool = new NativeWorkerPool(3, 4, 5, 5, "name", "id");
+        PassThroughTransportMetricsCollector metrics = new PassThroughTransportMetricsCollector(true, "testScheme");
+
+        TargetConfiguration targetConfiguration = new TargetConfiguration(configurationContext, null, workerPool,
+                metrics, null);
+        targetConfiguration.build();
+        HttpResponse response = PowerMockito.mock(HttpResponse.class, Mockito.RETURNS_DEEP_STUBS);
+        NHttpClientConnection conn = PowerMockito.mock(NHttpClientConnection.class, Mockito.RETURNS_DEEP_STUBS);
+        ContentDecoder decoder = PowerMockito.mock(ContentDecoder.class);
+        TargetConnections connections = PowerMockito.mock(TargetConnections.class);
+        targetConfiguration.setConnections(connections);
+
+        PowerMockito.mockStatic(TargetContext.class);
+
+        TargetContext cntxt = new TargetContext(targetConfiguration);
+        PowerMockito.when(TargetContext.get(any(NHttpClientConnection.class))).thenReturn(cntxt);
+        PowerMockito.when(decoder.read(any(ByteBuffer.class))).thenReturn(12);
```

Comment:
What is the magic number 12 here? Can you create a local variable with the meaningful name and use it in the logic?

Reply:
It's actually a random value. It's the value that is supposed to be returned by power mockito when a byte buffer is read.